plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -0,0 +1,251 @@
+     * @return all representations whose target is the specified EObject
+     */
+    private List<DRepresentation> getAssociatedRepresentationByName(EObject targetRootObject,
+            String diagramDescriptionName, Session session) {
+        List<DRepresentation> result = new ArrayList<DRepresentation>();
+        if (diagramDescriptionName != null) {
+            Collection<DRepresentation> representations = DialectManager.INSTANCE.getRepresentations(targetRootObject,
+                    session);
+
+            // Filter representations to keep only those in a selected viewpoint
+            Collection<Viewpoint> selectedViewpoints = session.getSelectedViewpoints(false);
+
+            for (DRepresentation representation : representations) {
+                if (representation instanceof DDiagram
+                    && diagramDescriptionName.equals(((DDiagram) representation).getDescription().getName())
+                    && representation.eContainer() instanceof DView) {
+                    DView dView = (DView) representation.eContainer();
+                    Viewpoint vp = dView.getViewpoint();
+                    if (selectedViewpoints.contains(vp)) {
+                        result.add(representation);
+                    }
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Get the GmF Diagram instance corresponding to a viewpoint
+     * DSemanticDiagram.
+     * 
+     * @param semanticDiagram
+     *            the diagram object from which we want to create an image.
+     * @return the GmF Diagram instance corresponding to a viewpoint
+     *         DSemanticDiagram.
+     */
+    private Diagram getGmfDiagram(DSemanticDiagram semanticDiagram) {
+        for (final AnnotationEntry annotation : new DDiagramQuery(semanticDiagram)
+                .getAnnotation(CustomDataConstants.GMF_DIAGRAMS)) {
+            EObject eObject = annotation.getData();
+            if (eObject instanceof Diagram) {
+                final Diagram diagramInResource = (Diagram) eObject;
+                final EObject semanticElement = ViewUtil.resolveSemanticElement(diagramInResource);
+                if (semanticElement.equals(semanticDiagram)) {
+                    return diagramInResource;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<String> getRepresentationImagePath(Map<String, Object> parameters) throws ProviderException {
+
+        EObject rootObject = (EObject) parameters.get(ProviderConstants.KEY_CONF_ROOT_OBJECT);
+        String rootPath = (String) parameters.get(ProviderConstants.KEY_PROJECT_ROOT_PATH);
+        Session session = SessionManager.INSTANCE.getSession(rootObject);
+        Object diagramDescriptionName = parameters.get(KEY_DIAGRAM_DESCRIPTION_NAME);
+        Object targetRootObject = parameters.get(KEY_TARGET_ROOT_OBJECT);
```

Comment:
It isn't useful to use both rootObject and targetRootObject. The targetRootObject should be sufficient in this case